regression/esbmc/github_2338_8/main.c

```diff
@@ -0,0 +1,8 @@
+#include <assert.h>
+
+int main() {
+  unsigned int a = 0x80000000; // Negative negation of the halfway point (overflow risk)
+  unsigned int result = -a; 
+  assert(result == 0x7FFFFFFF); // Incorrect expected result for unsigned negation
```

Comment:
I think this assert is never actually triggered, because the negation will cause an error beforehand.